py-polars/src/series/export.rs

```diff
@@ -357,14 +360,36 @@ where
     PyArray1::<T>::from_iter_bound(py, values).into_py(py)
 }
 /// Convert arrays by flattening first, converting the flat Series, and then reshaping.
-fn array_series_to_numpy(py: Python, s: &Series) -> PyObject {
+fn array_series_to_numpy(py: Python, s: &Series, writable: bool) -> PyObject {
     let ca = s.array().unwrap();
     let s_inner = ca.get_inner();
-    let np_array_flat = series_to_numpy_with_copy(py, &s_inner).unwrap();
+    let np_array_flat = series_to_numpy_with_copy(py, &s_inner, writable).unwrap();
 
     // Reshape to the original shape.
     let DataType::Array(_, width) = s.dtype() else {
         unreachable!()
     };
     reshape_numpy_array(py, np_array_flat, ca.len(), *width)
 }
+/// Convert lists by flattening first, converting the flat Series, and then splitting by offsets.
+fn list_series_to_numpy(py: Python, s: &Series, writable: bool) -> PyObject {
+    let ca = s.list().unwrap();
+    let s_inner = ca.get_inner();
+
+    let np_array_flat = series_to_numpy(py, &s_inner, writable, true).unwrap();
+
+    split_numpy_array(py, np_array_flat, ca)
+}
+fn split_numpy_array(py: Python, arr: PyObject, ca: &ListChunked) -> PyObject {
```

Comment:
Is this function ever intended to be used anywhere else? I think we should just merge this with the caller function. Especially since it accepts a `&ListChunked`.

I also think should be the slow path. If we have numeric values, we can create an owned `O`, `Arc` that `Arc<O>`, then take the slice `&[T]` and slice that in Rust and pass the rust slices and `Arc<O>::cloned()` to new numpy arrays. Then we don't have to slice in python.

Reply:
> Is this function ever intended to be used anywhere else? I think we should just merge this with the caller function. Especially since it accepts a &ListChunked.

Not really, I separated it mostly for readability. I improved the function signature take take the offsets as an iterator. If you still don't like this I'll move it into the `list_series_to_numpy` function.

> I also think should be the slow path. If we have numeric values, we can create an owned O, Arc that Arc<O>, then take the slice &[T] and slice that in Rust and pass the rust slices and Arc<O>::cloned() to new numpy arrays. Then we don't have to slice in python.

I don't fully understand but I think this involves downcasting the `PyObject` to a `PyArray`, right? That's not trivial because it could contain any type and be of any number of dimensions (in case of nested arrays). Not sure how to go about that. But I do see that calling a Python method in an iteration this way is not ideal...

Reply:
I inlined the split function and added a TODO for now.